njsscan/njsscan.py

```diff
@@ -101,6 +103,17 @@ def post_ignore_rules(self):
             if rule_id in self.result['templates']:
                 del self.result['templates'][rule_id]
 
+    def post_ignore_rules_by_severity(self):
```

Comment:
This method has to run for both `nodejs` and `templates` keys. We have results from both semgrep and pattern matcher and we handle it separately. I suggest calling the `post_ignore_rules_by_severity() method twice with `nodejs` and `template` as result keys.

Something like 

```python
def post_ignore_rules_by_severity(self, key):
        """Filter findings by rule severity."""
         del_keys = set()
         for rule_id, details in self.result[key].items():
             issue_severity = details.get('metadata').get('severity')
             if issue_severity not in self.options['severity_filter']:
                 del_keys.add(rule_id)
         for rid in del_keys:
             if rid in self.result[key]:
                 del self.result[key][rid]
 ```

Reply:
Corrected, did I hear you right?

Reply:
Perfect!